### PR TITLE
[ios][docs] Add support for options which allows displaying notifications in foreground

### PR DESCRIPTION
**Pull request by hesyifei**
# Why

https://expo.canny.io/feature-requests/p/implement-foreground-notifications-on-ios-using-new-native-willpresentnotificati

Currently, on iOS, all notifications received when the app is in foreground have to be handled by the developer separately. Since iOS 10, iOS supports also displaying the notification banner when the app is in foreground. This PR gives the developer an option to choose to let the system handles and displays the notification when the app is in foreground. (Ref: https://stackoverflow.com/a/40756206/2603230)

# How

In `userNotificationCenter` `willPresentNotification`, we first check if `notification.iosDisplayInForeground` in `app.json` is set to true. We also check if `_displayInForeground` is set for the individual message (which always override the prior option).

By defaults, `notification.iosDisplayInForeground` will be `false` (the same as before), so there will not be any breaking changes to the app's behavior.

In other words, the notification will be shown in foreground if:
- `notification.iosDisplayInForeground` is `true` and `_displayInForeground` is not `false`, or 
- `_displayInForeground` is `true`

# Test Plan

Different cases:

Each column is a value for `notification.iosDisplayInForeground` and each row is a value for `_displayInForeground`. A "✓" means the notification will be displayed in foreground.

|  | true | false | not set |
| ---: | --- | --- | --- |
| **true** | ✓ | ✓ | ✓ |
| **false** | ✗ | ✗ | ✗ |
| **not set** | ✓  | ✗ | ✗ (default) |

### CODE REVIEW

**Review by esamelson — APPROVED** [review]
looks great, thanks! I just have one question, which is: what is the behavior with local notifications? Do we also need to add a `displayInForeground` option to those, or do they already display in the foreground? I don't think that needs to block this PR from landing -- fine to just worry about pushes -- but let's at least create an issue to track it if we need to.

**Comment by hesyifei** [comment]
@esamelson Thanks for pointing that out! `willPresentNotification` is called for both local and push notifications. So the `notification.iosDisplayInForeground` property in `app.json` will be applied to both local and push notifications. I have also added `ios._displayInForeground` option to `LocalNotification`. (https://github.com/expo/expo/pull/4802/commits/8a5d402045769af5cd821b3eee1468c7591d7d87)

**Review by esamelson — APPROVED** [review]
awesome 👍 🙏 🎉
